### PR TITLE
API methods for working with scripts

### PR DESCRIPTION
**Pull request by oskin1**

### CODE REVIEW

**Comment by oskin1** [comment]
#526

File: src/main/scala/org/ergoplatform/api/WalletApiRoute.scala [hunk 1]
```diff
@@ -38,16 +42,38 @@ case class WalletApiRoute(readersHolder: ActorRef, nodeViewActorRef: ActorRef, e
...
   }
 
+  private val source: Directive1[String] = entity(as[Json]).flatMap { p =>
+    p.hcursor.downField("source").as[String]
+      .fold(_ => reject, s => provide(s))
+  }
+
+  private def addressResponse(address: ErgoAddress): Json = Json.obj("address" -> addressJsonEncoder(address))
+
+  private def compileSource(source: String, publicKeys: Seq[ProveDlog]): Try[Value[SBoolean.type]] = {
```
Comment (ergomorphic) [inline src/main/scala/org/ergoplatform/api/WalletApiRoute.scala]: @oskin1 why not allow env: Map[String, Any] in the place of publicKeys? Also what about ProveDHT?

Comment (oskin1) [reply src/main/scala/org/ergoplatform/api/WalletApiRoute.scala]: @ergomorphic, the idea is only to load available in wallet public keys into env, so why do we really need `env: Map[String, Any]` here?

Comment (oskin1) [reply src/main/scala/org/ergoplatform/api/WalletApiRoute.scala]: @ergomorphic, ok, I've fixed that.

File: src/test/scala/org/ergoplatform/utils/Stubs.scala [hunk 2]
```diff
@@ -134,12 +135,13 @@ trait Stubs extends ErgoGenerators with ErgoTestHelpers with ChainGenerator with
...
     private implicit val addressEncoder: ErgoAddressEncoder = new ErgoAddressEncoder(settings.chainSettings.addressPrefix)
     private val prover = new ErgoProvingInterpreter(seed, 2)
-    private val trackedAddresses: mutable.Buffer[ErgoAddress] =
-      mutable.Buffer(prover.dlogPubkeys: _ *).map(P2PKAddress.apply)
-
+    private val trackedAddresses: Seq[P2PKAddress] = prover.dlogPubkeys.map(P2PKAddress.apply)
 
     def receive: Receive = {
 
+      case ReadPublicKeys(from, until) =>
```
Comment (catena2w) [inline src/test/scala/org/ergoplatform/utils/Stubs.scala]: It is not used anywhere

Comment (oskin1) [reply src/test/scala/org/ergoplatform/utils/Stubs.scala]: @catena2w, it is required for tests as wallet route now read public keys from wallet, so we need stub to mock wallet actor response.
https://github.com/ergoplatform/ergo/blob/1db706f65805cac501f0464fe8674670e83a46ef/src/main/scala/org/ergoplatform/api/WalletApiRoute.scala#L144

**Pull request merged**